### ISSUE TITLE
GPL-633 Event warehouse should have same recovery and error handling of unified WH

### ISSUE THREAD

**Issue by JamesGlover** [sanger/event_warehouse#77]
**User story**
The event warehouse failed to auto-recover following database loss. This behaviour is already in place in the unified warehouse. We should probably extract out daemonization and error handing into its own gem, to ensure all our consumers function in the same way. This could unify consumers in:

- Event WH
- Unified Warehouse
- Sequencescape
- Warehouse 2

While the unified warehouse is probably most advanced feature wise, the Sequencescape consumers have an element of configurability that might be useful to extract, or at least take inspiration from.

**Who are the primary contacts for this story**
James G

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] add a list of acceptance criteria here
- [ ] ...

**Dependencies**
This story is blocked by the following dependencies:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

**Additional context**

**Comment by JamesGlover** [comment]
Gotcha: There is some custom dead-lettering going on in the event warehouse and warehouse 2, with a comment that the inbuilt dead-lettering is 'unreliable'. We *have* seen some production issues with dead-lettering which caused workers to stall, and attempts to localise the issue strongly suggested that it wasn't in or code.

Last time I saw these issues was back on the metal. I am not sure if they persist on the FCE.

**Comment by JamesGlover** [comment]
This is probably a case of pulling the shared handling out into warren

**Comment by JamesGlover** [comment]
Some examples:
Bingo AmqpConsumer: (196:"staging.event.sample_manifest.created.443374"): Dead lettering due to Mysql2::Error: MySQL client is not connected: BEGIN
So I know the event warehouse isn’t as clever as the unified warehouse in this case (Which manages to re-connect) but I thought they were supposed to panic and shut-down in this scenario. Also, looks like the dead-letter bindings may not be right, as I can’t see a queue.

**Comment by JamesGlover** [comment]
On hold pending update of warren functionality.
